### ISSUE TITLE
remotebackend can not find globalnames

### ISSUE THREAD

**Issue by przemas75** [polaris-gslb/polaris-gslb#20]
Hello,
I have tried polaris-gslb v. 0.4 with newest (git) pdns. With a config almost identical to an example i have these in debug output:

Jun 08 09:30:32 [remotebackend]: Polaris Remote Backend initialized request: {"method": "initialize", "parameters": {"command": "/opt/polaris/bin/polaris-pdns", "timeout": "2000"}} result: True pid: 29508 time taken: 0.000029
Jun 08 09:30:32 [remotebackend]: no globalname found for qname "www.example.com." request: {"method": "lookup", "parameters": {"local": "0.0.0.0", "qname": "www.example.com.", "qtype": "SOA", "real-remote": "192.168.51.202/32", "remote": "192.168.51.202", "zone-id": -1}} result: False pid: 29508 time taken: 0.001619
Jun 08 09:30:32 [remotebackend]: no globalname found for qname "example.com." request: {"method": "lookup", "parameters": {"local": "0.0.0.0", "qname": "example.com.", "qtype": "SOA", "real-remote": "192.168.51.202/32", "remote": "192.168.51.202", "zone-id": -1}} result: False pid: 29508 time taken: 0.000044
Jun 08 09:30:32 [remotebackend]: no globalname found for qname "com." request: {"method": "lookup", "parameters": {"local": "0.0.0.0", "qname": "com.", "qtype": "SOA", "real-remote": "192.168.51.202/32", "remote": "192.168.51.202", "zone-id": -1}} result: False pid: 29508 time taken: 0.000035
Jun 08 09:30:32 [remotebackend]: no globalname found for qname "." request: {"method": "lookup", "parameters": {"local": "0.0.0.0", "qname": ".", "qtype": "SOA", "real-remote": "192.168.51.202/32", "remote": "192.168.51.202", "zone-id": -1}} result: False pid: 29508 time taken: 0.000034

My polaris-lb.yaml looks like:
`pools:
    www:
      monitor: http
      monitor_params:
          use_ssl: false
          hostname: www.example.com
          url_path: /info.php
      lb_method: fogroup
      fallback: any
      members:
      - ip: 192.168.51.202
        name: debian-test
        weight: 1
      - ip: 192.168.99.101
        name: debian84
        weight: 1

globalnames:
    www.example.com:
        pool: www
        ttl: 1
`

This is debian jessie with some testing repo for new python:
python3
Python 3.5.1+ (default, May  9 2016, 11:00:17)
[GCC 5.3.1 20160429] on linux

**Comment by antongavrik** [comment]
What output do the below commands produce?
/opt/polaris/bin/polaris-memcache-control get-generic-state
/opt/polaris/bin/polaris-memcache-control get-ppdns-state

**Comment by przemas75** [comment]
Here it is:

```
 root@debiant:/var/log#  /opt/polaris/bin/polaris-memcache-control get-generic-state
{
    "globalnames": {
        "www.example.com": {
            "pool_name": "www",
            "ttl": 1,
            "name": "www.example.com"
        }
    },
    "timestamp": 1465218198.936368,
    "pools": {
        "www": {
            "max_addrs_returned": 1,
            "members": [
                {
                    "region": "None",
                    "retries_left": 2,
                    "ip": "192.168.51.202",
                    "status_reason": "monitor passed",
                    "weight": 1,
                    "name": "debian-test",
                    "last_probe_issued_time": 1465218197.9017353,
                    "status": true
                },
                {
                    "region": "None",
                    "retries_left": 2,
                    "ip": "192.168.99.101",
                    "status_reason": "monitor passed",
                    "weight": 1,
                    "name": "debian84",
                    "last_probe_issued_time": 1465218197.9017541,
                    "status": true
                }
            ],
            "lb_method": "fogroup",
            "monitor": {
                "timeout": 5,
                "url_path": "/info.php",
                "hostname": "www.example.com",
                "port": 80,
                "retries": 2,
                "interval": 10,
                "name": "http",
                "use_ssl": false
            },
            "last_status": true,
            "name": "www",
            "fallback": "any"
        }
    }
}
root@debiant:/var/log#  /opt/polaris/bin/polaris-memcache-control get-ppdns-state
{
    "globalnames": {
        "www.example.com": {
            "pool_name": "www",
            "ttl": 1
        }
    },
    "timestamp": 1465218198.934812,
    "pools": {
        "www": {
            "status": true,
            "dist_tables": {
                "_default": {
                    "rotation": [
                        "192.168.51.202"
                    ],
                    "index": 0,
                    "num_unique_addrs": 1
                }
            },
            "max_addrs_returned": 1,
            "lb_method": "fogroup",
            "fallback": "any"
        }
    }
}
root@debiant:/var/log#
```
This is installed on my system:
ii  memcached                              1.4.25-2                    amd64        high-performance memory object caching system

**Comment by antongavrik** [comment]
There is a trailing dot in the qname "www.example.com." sent by pdns, Polaris does not understand this.
What command are you using to perform the lookup? 
Another possibility is that there is a change in pdns json api behavior, please try a released version of pdns 3.4.8 or 3.4.9

**Comment by przemas75** [comment]
Indeed, with 3.4.9 version of PowerDNS polaris works  just fine.
Thanks.
